spec/index.bs

```diff
@@ -3197,11 +3201,28 @@ The following validation rules apply:
   For the copy ranges:
   - [=Valid Texture Copy Range=] applies to |source| and |copySize|.
   - [=Valid Texture Copy Range=] applies to |destination| and |copySize|.
-  - If |source|.{{GPUTextureCopyView/texture}} is the same as |destination|.{{GPUTextureCopyView/texture}}:
-    - The [=subresource=] at [=mipmap level=] |source|.{{GPUTextureCopyView/mipLevel}} and [=array layer=] |source|.{{GPUTextureCopyView/arrayLayer}} must be different from the [=subresource=] at [=mipmap level=] |destination|.{{GPUTextureCopyView/mipLevel}} and [=array layer=] |destination|.{{GPUTextureCopyView/arrayLayer}}.
+  - The [$set of subresources for texture copy$](|source|, |copySize|) and
+     the  [$set of subresources for texture copy$](|destination|, |copySize|) must be disjoint.
 
-</dfn>
+</div>
 
+<div algorithm>
+    The <dfn abstract-op>set of subresources for texture copy</dfn>(|textureCopyView|, |copySize|)
+    is the set containing:
+
+      - If |textureCopyView|.{{GPUTextureCopyView/texture}}.{{GPUTexture/[[dimension]]}}
+        is {{GPUTextureDimension/"2d"}}:
+          - For each |arrayLayer| of the |copySize|.[=Extent3D/depth=] [=array layers=]
+            starting at |textureCopyView|.{{GPUTextureCopyView/origin}}.[=Origin3D/z=]:
+              - The [=subresource=] of |textureCopyView|.{{GPUTextureCopyView/texture}} at
```

Comment:
I wonder if it's going to be cleaner to define a subresource in a function manner, i.e. `[$subresource$](texture, mipmapLevel, arrayLayer)`. This is just an idea, not something you have to fix here.